src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -594,9 +578,9 @@ public IReadOnlyList<Type> GetParameterTypes(string methodName)
                 // We use the parameter types of the first handler
                 lock (handlers)
                 {
-                    if (handlers.Count > 0)
+                    if (handlers.CopiedHandlers.Length > 0)
```

Comment:
Put this code (including the `lock` above) in a method on `InvocationHandlerList` and make sure it locks on `_invocationHandlers`. Because it locks on `handlers` here, it will actually be locking on a **different** object than `.Add` and `.Remove` (which lock on `handlers._invocationHandlers`), so it's not thread-safe